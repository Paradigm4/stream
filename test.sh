#!/bin/bash

MYDIR=`dirname $0`
pushd $MYDIR > /dev/null
MYDIR=`pwd`

iquery -aq "remove(foo)" > /dev/null 2>&1
rm -rf $MYDIR/test.out

iquery -anq "store(build(<val:double>[i=1:800000,100000,0], i), foo)" > /dev/null 2>&1
iquery -aq "op_count(stream(foo, '$MYDIR/examples/stream_test_client'))" >> $MYDIR/test.out 2>&1
iquery -aq "stream(foo, '$MYDIR/examples/stream_test_client')" | head -n 5 >> $MYDIR/test.out 2>&1
iquery -otsv -aq "stream(_sg(foo, 2,0), '$MYDIR/examples/stream_test_client SUMMARIZE')" | head -n 1 >> $MYDIR/test.out 2>&1

#Test READ_DELAY: start a client with a delay
iquery -aq "op_count(stream(foo, '$MYDIR/examples/stream_test_client READ_DELAY'))" > /dev/null 2>&1 &
sleep 1
#Find the query and cancel it
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1
QID=`iquery -otsv -aq "filter(list('queries'), inst=0 )" | grep stream | awk '{print $1}'`
iquery -aq "cancel('$QID')" >> $MYDIR/test.out 2>&1
sleep 5
#After 5 seconds, better be cleared up
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1

#Test WRITE_DELAY: start a client with a delay
iquery -aq "op_count(stream(foo, '$MYDIR/examples/stream_test_client WRITE_DELAY'))" > /dev/null 2>&1 &
sleep 1
#Find the query and cancel it
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1
QID=`iquery -otsv -aq "filter(list('queries'), inst=0 )" | grep stream | awk '{print $1}'`
iquery -aq "cancel('$QID')" >> $MYDIR/test.out 2>&1
sleep 5
#After 5 seconds, better be cleared up
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1

iquery -ocsv -aq "stream(build(<val:string> [i=1:10,5,0], i), 'Rscript $MYDIR/examples/tsv_R_client.R')" >> $MYDIR/test.out 2>&1
iquery -aq "op_count(stream(foo, 'Rscript $MYDIR/examples/tsv_R_client.R'))" >> $MYDIR/test.out 2>&1

iquery -aq "stream(build(<val:double> [i=1:5,5,0], i), 'Rscript $MYDIR/examples/R_client.R', 'format=df', 'types=double,int32')" >> $MYDIR/test.out 2>&1

iquery -aq "
 stream(
  _sg(
   stream(
    build(<val:double> [i=1:50,10,0], i), 
    'Rscript $MYDIR/examples/R_sum.R', 
    'format=df', 
    'types=double', 
    'names=sum'
   ), 
   2, 0
  ), 
  'Rscript $MYDIR/examples/R_sum.R', 
  'format=df', 
  'types=double', 
  'names=sum'
 )" >> $MYDIR/test.out 2>&1

iquery -aq "stream(build(<val:string> [i=1:10,10,0], iif(i=1,'', string(i))), 'Rscript $MYDIR/examples/R_strings.R', 'format=df', 'types=string', 'names=s')" >> $MYDIR/test.out 2>&1

iquery -aq "stream(
 apply(
  build(<a:int32>[i=1:3,3,0], i), 
  dub, double(iif(a=1, null, iif(a=2, 0,  1))),  
  i32, int32( iif(a=1, null, iif(a=2, 0,  1))), 
  str,        iif(a=1, null, iif(a=2, '', 'abc'))
 ), 
 'Rscript $MYDIR/examples/R_identity.R', 'format=df', 'types=int32,double,int32,string', 'names=a,b,c,d'
)" >> $MYDIR/test.out 2>&1

diff test.expected test.out

