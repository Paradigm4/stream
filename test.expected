{i} count
{0} 8
{instance_id,chunk_no} response
{0,0} 'Hello	1
Hello	2
Hello	3
Hello	4
Thanks! That was a total of 800000 lines.
{i} count
{0} 2
Query was executed successfully
{i} count
{0} 1
{i} count
{0} 2
Query was executed successfully
{i} count
{0} 1
'1	2
2	4
3	6
4	8
5	10'
'6	12
7	14
8	16
9	18
10	20'
{i} count
{0} 8
{instance_id,chunk_no,value_no} a0,a1
{0,0,0} 2,8
{0,0,1} 4,16
{0,0,2} 6,24
{0,0,3} 8,32
{0,0,4} 10,40
{instance_id,chunk_no,value_no} sum
{0,0,0} 1275
{instance_id,chunk_no,value_no} s
{0,0,0} 'Hello, !'
{0,0,1} 'Hello, 2!'
{0,0,2} 'Hello, 3!'
{0,0,3} 'Hello, 4!'
{0,0,4} 'Hello, 5!'
{0,0,5} 'Hello, 6!'
{0,0,6} 'Hello, 7!'
{0,0,7} 'Hello, 8!'
{0,0,8} 'Hello, 9!'
{0,0,9} 'Hello, 10!'
{0,0,10} ''
{0,0,11} 'Thx!'
{instance_id,chunk_no,value_no} a,b,c,d
{0,0,0} 1,null,null,null
{0,0,1} 2,0,0,''
{0,0,2} 3,1,1,'abc'
Hello\t\\n \\r \\t \\\\ \t\\N\nOK\tthanks!
{instance_id,chunk_no} response
{0,0} 'I got	0
I got	1
I got	2
I got	3
I got	4
I got	5
I got	6
I got	7
I got	8
I got	9
THX!'
{0,1} 'KTHXBYE'
